### ISSUE TITLE
SbdhWrapper does not work as expected

### ISSUE THREAD

**Issue by monze** [difi/oxalis#239]
We are using an older version of Oxalis, but the problem still exist in current trunk.
When we send a document, it sometimes happens that we get back this error message:

AS2 transmission failed : The following headers were received: date: on, 18 nov. 2015 12:24:08 +0000 message-id: ...... The message sent to AS2 System id APP_1000000066 on on, 18 nov. 2015 12:24:06 +0000 with subject AS2 MDN as you requested has been received. It has been processed The warning/error message is : ERROR: Payload does not contain Standard Business Document Header

Note that the error is: Payload does not contain Standard Business Document Header

We newer create the SBDH, and therefore relay on Oxalis can creating the SBDH for us. This is performed in the class SbdhWrapper.java. However, if the creation of the SBDH fails, the error is just ignored, and the document is attempted to transmitted anyway.

```java
        } catch (Exception ex) {
            ex.printStackTrace();
        }
```

I would have expected that Oxalis would stop transmitting instead of continue. Oxalis has already decided that the document is without SBDH, and Oxalis was unable to create a SBDH.

https://github.com/difi/oxalis/blob/master/oxalis-commons/src/main/java/eu/peppol/document/SbdhWrapper.java

I have not yet discovered why the SbdhWrapper failes, that is my next step. I know that it can failed, if we are used the wrong encoding, and thereby some special invalid characters.

Jacob Mogensen
mySupply ApS

**Comment by monze** [comment]
I have found the reason, why we failed in sending, 

But the problem in SbdhWrapper.java still exist, and should be fix'ed.

My problem was the position of the reader in the input stream was not resat to zero (I have copy the content of the inputStream to a fileOutputStream, to validate that no encoding error was present.)
Therefore this method resulted in a payload that consist of a byte array consisting of 4 * 1024 empty bytes.

TransmissionRequestBuilderWithoutParsing.java
```
    protected void savePayLoad(InputStream inputStream) {
        try {
            payload = Util.intoBuffer(inputStream, 101L * 1024 * 1024);     // Copies the contents into a buffer
        } catch (IOException e) {
            throw new IllegalStateException("Unable to save the payload: " + e.getMessage(), e);
        }
    }
```

Again, the problem in SbdhWrapper.java still exist, and should be fix'ed.

Jacob Mogensen
mySupply ApS
